crates/turbopath/src/absolute_system_path_buf.rs

```diff
@@ -0,0 +1,175 @@
+        Ok(AbsoluteSystemPathBuf(system_path))
+    }
+
+    /// Converts `path` to an `AbsoluteSystemPathBuf` without validating that
+    /// it is either absolute or a system path.
+    ///
+    /// # Arguments
+    ///
+    /// * `path`: The path to be converted to an `AbsoluteSystemPathBuf`.
+    ///
+    /// returns: AbsoluteSystemPathBuf
+    pub fn new_unchecked(path: impl Into<PathBuf>) -> Self {
+        let path = path.into();
+        AbsoluteSystemPathBuf(path)
+    }
+
+    /// Anchors `path` at `self`.
+    ///
+    /// # Arguments
+    ///
+    /// * `path`: The path to be anchored at `self`
+    ///
+    /// returns: Result<AnchoredSystemPathBuf, PathValidationError>
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use std::path::Path;
+    /// use turbopath::{AbsoluteSystemPathBuf, AnchoredSystemPathBuf};
+    /// let base = AbsoluteSystemPathBuf::new("/Users/user").unwrap();
+    /// let anchored_path = AbsoluteSystemPathBuf::new("/Users/user/Documents").unwrap();
+    /// let anchored_path = base.anchor(&anchored_path).unwrap();
+    /// assert_eq!(anchored_path.as_path(), Path::new("Documents"));
+    /// ```
+    pub fn anchor(
+        &self,
+        path: &AbsoluteSystemPathBuf,
+    ) -> Result<AnchoredSystemPathBuf, PathValidationError> {
+        AnchoredSystemPathBuf::strip_root(self, path)
+    }
+
+    /// Resolves `path` with `self` as anchor.
+    ///
+    /// # Arguments
+    ///
+    /// * `path`: The path to be anchored at `self`
+    ///
+    /// returns: AbsoluteSystemPathBuf
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use std::path::Path;
+    /// use turbopath::{AbsoluteSystemPathBuf, AnchoredSystemPathBuf};
+    /// let absolute_path = AbsoluteSystemPathBuf::new("/Users/user").unwrap();
+    /// let anchored_path = AnchoredSystemPathBuf::new_unchecked("Documents");
+    /// let resolved_path = absolute_path.resolve(&anchored_path);
+    /// #[cfg(windows)]
+    /// assert_eq!(resolved_path.as_path(), Path::new("\\Users\\user\\Documents"));
+    /// assert_eq!(resolved_path.as_path(), Path::new("/Users/user/Documents"));
+    /// ```
+    pub fn resolve(&self, path: &AnchoredSystemPathBuf) -> AbsoluteSystemPathBuf {
+        AbsoluteSystemPathBuf(self.0.join(path.as_path()))
+    }
+
+    pub fn as_path(&self) -> &Path {
+        self.0.as_path()
+    }
+
+    pub fn components(&self) -> Components<'_> {
+        self.0.components()
+    }
+
+    pub fn parent(&self) -> Option<Self> {
+        self.0
+            .parent()
+            .map(|p| AbsoluteSystemPathBuf(p.to_path_buf()))
+    }
+
+    pub fn starts_with<P: AsRef<Path>>(&self, base: P) -> bool {
+        self.0.starts_with(base.as_ref())
+    }
+
+    pub fn ends_with<P: AsRef<Path>>(&self, child: P) -> bool {
+        self.0.ends_with(child.as_ref())
+    }
+
+    pub fn join<P: AsRef<Path>>(&self, path: P) -> AbsoluteSystemPathBuf {
+        AbsoluteSystemPathBuf(self.0.join(path))
+    }
```

Comment:
Do we want to expose this method? It has (IMO) non-obvious behavior e.g. `AbsoluteSystemPathBuf::new("/tmp/foo").unwrap().join("/etc") == "/etc"`

Reply:
Eh, that behavior definitely doesn't seem great. I think there is likely a place for `join_literal` or `join_unix_path` or `join_tokens` that takes a slice or something. But none of those should clobber the underlying absolute path.